test/UnitTest/Microsoft.AspNet.OData.Test.Shared/OpenComplexTypeTest.cs

```diff
@@ -161,6 +167,102 @@ public class OpenComplexTypeTest
             Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
         }
 
+        [Fact]
+        public async Task OpenComplexType_PatchNestedComplexTypeProperty()
+        {
+            // Arrange
+            string payload = @"{
+                ""Street"":""UpdatedStreet"",
+                ""Token@odata.type"":""#Guid"",
+                ""Token"":""9B198CA0-9546-4162-A4C0-14EAA255ACA7"",
+                ""BirthDay@odata.type"":""#Date"",
+                ""BirthDay"":""2016-01-29"",
+                ""LineB"": {""Description"": ""DescriptionB""}
+            }";
+            payload = Regex.Replace(payload, @"\s*", "", RegexOptions.Multiline);
```

Comment:
>Regex. [](start = 22, length = 6)

I remember there's an email related to Regex.

I can't remember the detail. But, it seems this lib is not recommended to us.

Reply:
I think you are referring to regex's "catastrophic regex backtracking" problem that could cause CPU starvation and DOS. This is test code, and I don't think above pattern is backtracking pattern. So we should be able to use Regex here.

---
In reply to: [190751713](https://github.com/OData/WebApi/pull/1445#discussion_r190751713) [](ancestors = 190751713)